plugins/package-managers/pub/src/main/kotlin/Pub.kt

```diff
@@ -450,11 +457,16 @@ class Pub(
                 "Analyzing Android dependencies for package '$packageName' using Gradle version $pubGradleVersion."
             }
 
-            val gradleAnalyzerConfig = analyzerConfig.withPackageManagerOption(
-                gradleFactory.type,
-                OPTION_GRADLE_VERSION,
-                pubGradleVersion
-            )
+            val gradleAnalyzerConfig = analyzerConfig
+                .withPackageManagerOption(gradleFactory.type, OPTION_GRADLE_VERSION, pubGradleVersion).apply {
+                    options[OPTION_GRADLE_JAVA_VERSION]?.also { javaVersion ->
+                        withPackageManagerOption(gradleFactory.type, OPTION_GRADLE_JAVA_VERSION, javaVersion)
+                    }
+
+                    options[OPTION_GRADLE_JAVA_HOME]?.also { javaHome ->
+                        withPackageManagerOption(gradleFactory.type, OPTION_GRADLE_JAVA_HOME, javaHome)
+                    }
+                }
```

Comment:
If these option are configured for the `GradleInspector`, the factory call below should already apply them. Did you consider this?

Reply:
Ah, you're right. I can probably solve my problem in the context of running `PubFunTest` on Java 21, where "resolve dependencies for a project with Flutter, Android and Cocoapods" would otherwise fail as it does not set a lower Java version for the Gradle projects, differently.